src/types.ts

```diff
@@ -61,11 +61,11 @@ export interface InputObject {
   $schema?: Schema;
   $resolve?: ResolveFn;
   $default?: any;
-  [key: string]: any;
+  [key: string]: JSValue | InputObject;
 }
 
 export type InputValue = InputObject | JSValue;
 
 export type SchemaDefinition = {
-  [x: string]: JSValue | InputObject | SchemaDefinition;
+  [x: string]: JSValue | InputObject;
```

Comment:
We support recursive schema definition I suppose. (Inside a schema there can be mix of input and additional schemas)

Reply:
I think the type still works as before, as InputObject is identical to SchemaDefinition with the exception of certain reserved keys.

Reply:
I trust you then :)